### DISCUSSION TITLE
Question about pyi file

### DISCUSSION THREAD

**Discussion by jgarte** [Textualize/textual#2012] [Q&A] [answered]
Hi,

Why is this declaration required?

Seems redundant but just trying to understand:

https://github.com/Textualize/textual/blob/503c03730ed2fb2029925aa43623dde1cd0b9600/src/textual/widgets/__init__.pyi

**Comment by willmcgugan** [comment, marked as answer]
We have a system that lazy loads modules, so you can import one or two widgets without importing everything. Works great, but it broken typing. The `.pyi` allows type checkers to understand the types without the lazy loading getting in the way.
